bundles/org.openhab.binding.hive/src/main/java/org/openhab/binding/hive/internal/handler/HiveBridgeHandler.java

```diff
@@ -0,0 +1,394 @@
+                return reading;
+            }
+            try {
+                // Get battery level
+                response = client
+                        .newRequest("https://api-prod.bgchprod.info:443/omnia/nodes/"
+                                + thing.getProperties().get("linkedDevice"))
+                        .method(HttpMethod.GET).header("Accept", "application/vnd.alertme.zoo-6.1+json")
+                        .header("Content-Type", "application/vnd.alertme.zoo-6.1+json")
+                        .header("X-Omnia-Client", "Openhab 2").header("X-Omnia-Access-Token", token)
+                        .timeout(DISCOVER_TIMEOUT_SECONDS, TimeUnit.SECONDS).send();
+
+                statusCode = response.getStatus();
+                if (statusCode != HttpStatus.OK_200) {
+                    // If it failed, log the error
+                    String statusLine = response.getStatus() + " " + response.getReason();
+                    logger.warn("Error while reading from Hive API: {}", statusLine);
+                    return reading;
+                }
+                responseString = response.getContentAsString();
+                o = gson.fromJson(responseString, HiveNodes.class);
+            } catch (InterruptedException | TimeoutException | ExecutionException e) {
+                logger.warn("Failed to get battery leve: {}", e.getMessage());
+            }
+
+            reading.batteryLevel = o.nodes.get(0).attributes.batteryLevel;
+            return reading;
+        }
+        return reading;
+    }
+
+    private void callClient(String setObject, ThingUID uid, String type) {
+        try {
+            ContentResponse response;
+            response = client.newRequest("https://api-prod.bgchprod.info:443/omnia/nodes/" + uid.getId())
+                    .method(HttpMethod.PUT).header("Accept", "application/vnd.alertme.zoo-6.1+json")
+                    .header("Content-Type", "application/vnd.alertme.zoo-6.1+json")
+                    .header("X-Omnia-Client", "Openhab 2").header("X-Omnia-Access-Token", token)
+                    .timeout(DISCOVER_TIMEOUT_SECONDS, TimeUnit.SECONDS).content(new StringContentProvider(setObject))
+                    .send();
+
+            int statusCode = response.getStatus();
+
+            if (statusCode == HttpStatus.UNAUTHORIZED_401) {
+                // Token expired, get a new one and try again
+                getToken();
+                statusCode = response.getStatus();
+            }
+
+            if (statusCode != HttpStatus.OK_200) {
+                // If it failed, log the error
+                String statusLine = response.getStatus() + " " + response.getReason();
+                logger.warn("Error while reading from Hive API: {}", statusLine);
+            }
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+            logger.warn("Failed to update {}: {}", type, e.getMessage());
```

Comment:
If calls like this fail (repeatedly) its time to update the Thing Status, logging is not enough